### PR TITLE
feat: support build metadata

### PR DESCRIPTION
**Pull request by gregberge**
In order to avoid using a build if some tests have failed.

### CODE REVIEW

File: apps/backend/src/build/finalizeBuild.ts [hunk 1]
```diff
@@ -1,18 +1,122 @@
...
+
+/**
+ * Check if the bucket is valid from the metadata.
+ */
+export function checkIsBucketValidFromMetadata(metadata: BuildMetadata | null) {
+  return !metadata?.testReport || metadata.testReport.status === "passed";
+}
+
+function add(a: number | undefined, b: number | undefined) {
+  if (!a) {
```
Comment (jsfez) [inline apps/backend/src/build/finalizeBuild.ts:14]: missing `if(!a && !b)` case

File: apps/backend/src/build/finalizeBuild.ts [hunk 2]
```diff
@@ -1,18 +1,122 @@
...
+      if (!acc.testReport?.stats?.startTime) {
+        return item.testReport?.stats?.startTime;
+      }
+      if (!item.testReport?.stats?.startTime) {
+        return acc.testReport.stats.startTime;
+      }
+      return item.testReport.stats.startTime < acc.testReport.stats.startTime
+        ? item.testReport.stats.startTime
+        : acc.testReport.stats.startTime;
+    })();
```
Comment (jsfez) [inline apps/backend/src/build/finalizeBuild.ts:56]:
```suggestion
  return allMetatada.reduce<BuildMetadata | null>((acc, item) => {
    if (!item?.testReport) return acc;
    if (!acc) return item; 
    
    const status = acc.testReport.status === "failed" || item.testReport.status !== "passed"
      ? "failed"
      : "passed";

    const startTime = (() => {
      const accStartTime = acc.testReport?.stats?.startTime;
      const itemStartTime = item.testReport?.stats?.startTime;
      
      if (!accStartTime) return itemStartTime;
      if (!itemStartTime) return accStartTime;
      
      return Math.min(accStartTime, itemStartTime);
    })();
```

File: apps/backend/src/build/finalizeBuild.ts (hunk 1 above)
Comment (gregberge) [reply apps/backend/src/build/finalizeBuild.ts:14]: Not needed.

**Review by jsfez — APPROVED** [review]

File: apps/backend/src/build/finalizeBuild.ts (hunk 2 above)
Comment (gregberge) [reply apps/backend/src/build/finalizeBuild.ts:56]: It's 100% equivalent right?

Comment (jsfez) [reply apps/backend/src/build/finalizeBuild.ts:56]: Yes, it’s a style improvement suggestion.

**Pull request merged**